poetry/console/commands/shell.py

```diff
@@ -0,0 +1,55 @@
+from os import environ
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        from poetry.config import Config
+
+        # Check if it's already activated or doesn't exist and won't be created
+        if "POETRY_ACTIVE" in environ or not self.venv.is_venv():
+            current_venv = environ.get("VIRTUAL_ENV")
+
+            if current_venv:
+                self.line(
+                    "Virtual environment already activated: "
+                    "<info>{}</>".format(current_venv)
+                )
+
+            else:
+                config = Config.create("config.toml")
+                if not config.setting("settings.virtualenvs.create"):
```

Comment:
Is it possible that this `if` will be `False`? what then?

Reply:
Only if the virtualenv is supposed to be created, but for some reason wasn't. Maybe I don't even need to import `Config` and if it gets to this `else` block, it just gives that error message ("Virtual environment not found"). What do you think?

Reply:
Yep, I think that would be more concise. Eventually you could provide more verbose error message, but since it's getting late in Warsaw I can't come up with a satisfying proposition right now ;)